PurrSoft/PurrSoft.Api/Controllers/AnimalController.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using PurrSoft.Api.Controllers.Base;
+using PurrSoft.Application.Commands.AnimalCommands;
+using PurrSoft.Application.Common;
+using PurrSoft.Domain.Entities;
+using System.Net;
+
+namespace PurrSoft.Api.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class AnimalController : BaseController
+{
+    public AnimalController()
+    {
+    }
+
+    [HttpGet]
+    [Authorize(AuthenticationSchemes = "Bearer")]
+    [ProducesResponseType(typeof(ICollection<Animal>), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(CommandResponse), (int)HttpStatusCode.BadRequest)]
+    public async Task<CollectionResponse<Animal>> GetAnimalsAsync()
+    {
+        CollectionResponse<Animal> commandResponse =
+            await Mediator.Send(new AnimalGetCommand());
+
+        return commandResponse;
+    }
+
+    [HttpPost]
+    [Authorize(AuthenticationSchemes = "Bearer")]
+    [ProducesResponseType(typeof(CommandResponse<int>), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(CommandResponse), (int)HttpStatusCode.BadRequest)]
+    public async Task<IActionResult> CreateAnimalAsync(AnimalCreateCommand animalCreateCommand)
```

Comment:
Can anyone who is logged in create/update animals. Update maybe but create? Or can only the manager. You can Authorize only the managers if so for Create. 
Or even Delete.